init/eessi_environment_variables

```diff
@@ -25,9 +25,15 @@ if [ -d $EESSI_PREFIX ]; then
   if [ -d $EESSI_EPREFIX ]; then
 
     # determine subdirectory in software layer
-    # note: eessi_software_subdir_for_host.py will pick up value from $EESSI_SOFTWARE_SUBDIR_OVERRIDE if it's defined!
-    export EESSI_EPREFIX_PYTHON=$EESSI_EPREFIX/usr/bin/python3
-    export EESSI_SOFTWARE_SUBDIR=$($EESSI_EPREFIX_PYTHON ${EESSI_INIT_DIR_PATH}/eessi_software_subdir_for_host.py $EESSI_PREFIX)
+    if [ -z $EESSI_USE_ARCHDETECT ]; then
+      # if archdetect is enabled, use internal code
+      export EESSI_SOFTWARE_SUBDIR=$(${EESSI_INIT_DIR_PATH}/eessi_archdetect.sh cpupath)
+      echo "archdetect says ${EESSI_SOFTWARE_SUBDIR}"
```

Comment:
If we have this `echo`, we should also have it for `archspec` below, so it's always clear which mode is used

Reply:
done